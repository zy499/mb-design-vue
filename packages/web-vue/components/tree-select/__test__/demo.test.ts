import demoTest from '../../../scripts/demo-test';

demoTest('tree-select');
