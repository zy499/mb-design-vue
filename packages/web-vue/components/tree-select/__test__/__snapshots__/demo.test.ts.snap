// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<tree-select> demo: render [basic] correctly 1`] = `
"<span class=\\"mb-select-view-single mb-select-view mb-select-view-size-medium\\" border=\\"false\\" style=\\"width: 300px;\\"><!----><input class=\\"mb-select-view-input\\" readonly=\\"\\" placeholder=\\"Please select ...\\"><span class=\\"mb-select-view-value mb-select-view-value-hidden\\"><!----></span><span class=\\"mb-select-view-suffix\\"><!----><span class=\\"mb-select-view-icon\\"><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"mb-icon mb-icon-down mb-select-view-arrow-icon\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\"><path d=\\"M39.6 17.443 24.043 33 8.487 17.443\\"></path></svg></span>
<!----></span></span>
<!---->"
`;

exports[`<tree-select> demo: render [checkable] correctly 1`] = `
"<div style=\\"margin-bottom: 24px;\\"><label class=\\"mb-checkbox\\"><input type=\\"checkbox\\" class=\\"mb-checkbox-target\\"><span class=\\"mb-icon-hover mb-checkbox-icon-hover\\"><div class=\\"mb-checkbox-icon\\"><!----></div></span><span class=\\"mb-checkbox-label\\"> treeCheckStrictly </span></label></div><span class=\\"mb-select-view mb-select-view-size-medium mb-select-view-has-suffix mb-select-view-has-placeholder mb-select-view-multiple\\" border=\\"false\\" style=\\"width: 300px;\\"><span class=\\"mb-select-view-mirror\\">Please select ...</span>
<!---->
<transition-group-stub><input class=\\"mb-select-view-input\\" style=\\"width: 12px;\\" placeholder=\\"Please select ...\\"></transition-group-stub>
<!----><span class=\\"mb-select-view-suffix\\"><!----><span class=\\"mb-select-view-icon\\"><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"mb-icon mb-icon-expand\\" style=\\"transform: rotate(-45deg);\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\"><path d=\\"M7 26v14c0 .552.444 1 .996 1H22m19-19V8c0-.552-.444-1-.996-1H26\\"></path></svg></span>
<!---->
<!----></span></span>
<!---->"
`;

exports[`<tree-select> demo: render [checked-strategy] correctly 1`] = `
"<div style=\\"margin-bottom: 24px;\\"><span class=\\"mb-radio-group-button mb-radio-group-size-undefined mb-radio-group-direction-horizontal\\"><label class=\\"mb-radio-button mb-radio-checked\\"><input type=\\"radio\\" class=\\"mb-radio-target\\" value=\\"all\\"><span class=\\"mb-radio-button-content\\">show all</span></label><label class=\\"mb-radio-button\\"><input type=\\"radio\\" class=\\"mb-radio-target\\" value=\\"parent\\"><span class=\\"mb-radio-button-content\\">show parent</span></label><label class=\\"mb-radio-button\\"><input type=\\"radio\\" class=\\"mb-radio-target\\" value=\\"child\\"><span class=\\"mb-radio-button-content\\">show child</span></label></span></div><span class=\\"mb-select-view mb-select-view-size-medium mb-select-view-has-suffix mb-select-view-has-placeholder mb-select-view-multiple\\" border=\\"false\\" style=\\"width: 300px;\\"><span class=\\"mb-select-view-mirror\\">Please select ...</span>
<!---->
<transition-group-stub><input class=\\"mb-select-view-input\\" style=\\"width: 12px;\\" placeholder=\\"Please select ...\\"></transition-group-stub>
<!----><span class=\\"mb-select-view-suffix\\"><!----><span class=\\"mb-select-view-icon\\"><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"mb-icon mb-icon-expand\\" style=\\"transform: rotate(-45deg);\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\"><path d=\\"M7 26v14c0 .552.444 1 .996 1H22m19-19V8c0-.552-.444-1-.996-1H26\\"></path></svg></span>
<!---->
<!----></span></span>
<!---->"
`;

exports[`<tree-select> demo: render [control] correctly 1`] = `
"<span class=\\"mb-select-view-single mb-select-view mb-select-view-size-medium\\" border=\\"false\\" style=\\"width: 300px;\\"><!----><input class=\\"mb-select-view-input mb-select-view-input-hidden\\" readonly=\\"\\" placeholder=\\"Please select ...\\"><span class=\\"mb-select-view-value\\">Leaf</span><span class=\\"mb-select-view-suffix\\"><!----><span class=\\"mb-select-view-icon\\"><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"mb-icon mb-icon-down mb-select-view-arrow-icon\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\"><path d=\\"M39.6 17.443 24.043 33 8.487 17.443\\"></path></svg></span>
<!----></span></span>
<!---->"
`;

exports[`<tree-select> demo: render [fallback] correctly 1`] = `
"<div class=\\"mb-space mb-space-vertical\\">
  <div class=\\"mb-space-item\\" style=\\"margin-bottom: 24px;\\"><span class=\\"mb-select-view-single mb-select-view mb-select-view-size-medium\\" border=\\"false\\" style=\\"width: 300px;\\"><!----><input class=\\"mb-select-view-input mb-select-view-input-hidden\\" readonly=\\"\\" placeholder=\\"Please select ...\\"><span class=\\"mb-select-view-value\\">node0</span><span class=\\"mb-select-view-suffix\\"><!----><span class=\\"mb-select-view-icon\\"><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"mb-icon mb-icon-down mb-select-view-arrow-icon\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\"><path d=\\"M39.6 17.443 24.043 33 8.487 17.443\\"></path></svg></span>
    <!----></span></span>
    <!---->
  </div>
  <div class=\\"mb-space-item\\" style=\\"margin-bottom: 24px;\\"><span class=\\"mb-select-view-single mb-select-view mb-select-view-size-medium\\" border=\\"false\\" style=\\"width: 300px;\\"><!----><input class=\\"mb-select-view-input\\" readonly=\\"\\" placeholder=\\"Please select ...\\"><span class=\\"mb-select-view-value mb-select-view-value-hidden\\"><!----></span><span class=\\"mb-select-view-suffix\\"><!----><span class=\\"mb-select-view-icon\\"><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"mb-icon mb-icon-down mb-select-view-arrow-icon\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\"><path d=\\"M39.6 17.443 24.043 33 8.487 17.443\\"></path></svg></span>
    <!----></span></span>
    <!---->
  </div>
  <div class=\\"mb-space-item\\" style=\\"margin-bottom: 24px;\\"><span class=\\"mb-select-view-single mb-select-view mb-select-view-size-medium\\" border=\\"false\\" style=\\"width: 300px;\\"><!----><input class=\\"mb-select-view-input mb-select-view-input-hidden\\" readonly=\\"\\" placeholder=\\"Please select ...\\"><span class=\\"mb-select-view-value\\">++node0++</span><span class=\\"mb-select-view-suffix\\"><!----><span class=\\"mb-select-view-icon\\"><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"mb-icon mb-icon-down mb-select-view-arrow-icon\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\"><path d=\\"M39.6 17.443 24.043 33 8.487 17.443\\"></path></svg></span>
    <!----></span></span>
    <!---->
  </div>
  <div class=\\"mb-space-item\\" style=\\"margin-bottom: 24px;\\"><span class=\\"mb-select-view mb-select-view-size-medium mb-select-view-disabled-input mb-select-view-has-tag mb-select-view-has-suffix mb-select-view-multiple\\" border=\\"false\\" style=\\"width: 300px;\\"><span class=\\"mb-select-view-mirror\\"></span>
    <!---->
    <transition-group-stub><span class=\\"mb-tag mb-tag-size-medium mb-tag-checked mb-select-view-tag\\"><!--v-if-->node0<span class=\\"mb-icon-hover mb-tag-icon-hover mb-tag-close-btn\\" role=\\"button\\" aria-label=\\"Close\\"><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"mb-icon mb-icon-close\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\"><path d=\\"M9.857 9.858 24 24m0 0 14.142 14.142M24 24 38.142 9.858M24 24 9.857 38.142\\"></path></svg></span>
      <!--v-if--></span><span class=\\"mb-tag mb-tag-size-medium mb-tag-checked mb-select-view-tag\\"><!--v-if-->Leaf1<span class=\\"mb-icon-hover mb-tag-icon-hover mb-tag-close-btn\\" role=\\"button\\" aria-label=\\"Close\\"><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"mb-icon mb-icon-close\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\"><path d=\\"M9.857 9.858 24 24m0 0 14.142 14.142M24 24 38.142 9.858M24 24 9.857 38.142\\"></path></svg></span>
      <!--v-if--></span><input class=\\"mb-select-view-input\\" style=\\"width: 12px;\\" readonly=\\"\\">
    </transition-group-stub>
    <!----><span class=\\"mb-select-view-suffix\\"><!----><span class=\\"mb-select-view-icon\\"><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"mb-icon mb-icon-expand\\" style=\\"transform: rotate(-45deg);\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\"><path d=\\"M7 26v14c0 .552.444 1 .996 1H22m19-19V8c0-.552-.444-1-.996-1H26\\"></path></svg></span>
    <!---->
    <!----></span></span>
    <!---->
  </div>
  <div class=\\"mb-space-item\\" style=\\"margin-bottom: 24px;\\"><span class=\\"mb-select-view mb-select-view-size-medium mb-select-view-disabled-input mb-select-view-has-tag mb-select-view-has-suffix mb-select-view-multiple\\" border=\\"false\\" style=\\"width: 300px;\\"><span class=\\"mb-select-view-mirror\\"></span>
    <!---->
    <transition-group-stub><span class=\\"mb-tag mb-tag-size-medium mb-tag-checked mb-select-view-tag\\"><!--v-if-->Leaf1<span class=\\"mb-icon-hover mb-tag-icon-hover mb-tag-close-btn\\" role=\\"button\\" aria-label=\\"Close\\"><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"mb-icon mb-icon-close\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\"><path d=\\"M9.857 9.858 24 24m0 0 14.142 14.142M24 24 38.142 9.858M24 24 9.857 38.142\\"></path></svg></span>
      <!--v-if--></span><input class=\\"mb-select-view-input\\" style=\\"width: 12px;\\" readonly=\\"\\">
    </transition-group-stub>
    <!----><span class=\\"mb-select-view-suffix\\"><!----><span class=\\"mb-select-view-icon\\"><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"mb-icon mb-icon-expand\\" style=\\"transform: rotate(-45deg);\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\"><path d=\\"M7 26v14c0 .552.444 1 .996 1H22m19-19V8c0-.552-.444-1-.996-1H26\\"></path></svg></span>
    <!---->
    <!----></span></span>
    <!---->
  </div>
  <div class=\\"mb-space-item\\"><span class=\\"mb-select-view mb-select-view-size-medium mb-select-view-disabled-input mb-select-view-has-tag mb-select-view-has-suffix mb-select-view-multiple\\" border=\\"false\\" style=\\"width: 300px;\\"><span class=\\"mb-select-view-mirror\\"></span>
    <!---->
    <transition-group-stub><span class=\\"mb-tag mb-tag-size-medium mb-tag-checked mb-select-view-tag\\"><!--v-if-->++node0++<span class=\\"mb-icon-hover mb-tag-icon-hover mb-tag-close-btn\\" role=\\"button\\" aria-label=\\"Close\\"><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"mb-icon mb-icon-close\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\"><path d=\\"M9.857 9.858 24 24m0 0 14.142 14.142M24 24 38.142 9.858M24 24 9.857 38.142\\"></path></svg></span>
      <!--v-if--></span><span class=\\"mb-tag mb-tag-size-medium mb-tag-checked mb-select-view-tag\\"><!--v-if-->Leaf1<span class=\\"mb-icon-hover mb-tag-icon-hover mb-tag-close-btn\\" role=\\"button\\" aria-label=\\"Close\\"><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"mb-icon mb-icon-close\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\"><path d=\\"M9.857 9.858 24 24m0 0 14.142 14.142M24 24 38.142 9.858M24 24 9.857 38.142\\"></path></svg></span>
      <!--v-if--></span><input class=\\"mb-select-view-input\\" style=\\"width: 12px;\\" readonly=\\"\\">
    </transition-group-stub>
    <!----><span class=\\"mb-select-view-suffix\\"><!----><span class=\\"mb-select-view-icon\\"><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"mb-icon mb-icon-expand\\" style=\\"transform: rotate(-45deg);\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\"><path d=\\"M7 26v14c0 .552.444 1 .996 1H22m19-19V8c0-.552-.444-1-.996-1H26\\"></path></svg></span>
    <!---->
    <!----></span></span>
    <!---->
  </div>
</div>"
`;

exports[`<tree-select> demo: render [field-names] correctly 1`] = `
"<span class=\\"mb-select-view-single mb-select-view mb-select-view-size-medium\\" border=\\"false\\" style=\\"width: 300px;\\"><!----><input class=\\"mb-select-view-input mb-select-view-input-hidden\\" readonly=\\"\\" placeholder=\\"Please select ...\\"><span class=\\"mb-select-view-value\\">0-0-1</span><span class=\\"mb-select-view-suffix\\"><!----><span class=\\"mb-select-view-icon\\"><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"mb-icon mb-icon-down mb-select-view-arrow-icon\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\"><path d=\\"M39.6 17.443 24.043 33 8.487 17.443\\"></path></svg></span>
<!----></span></span>
<!---->"
`;

exports[`<tree-select> demo: render [label-in-value] correctly 1`] = `
"<span class=\\"mb-select-view-single mb-select-view mb-select-view-size-medium\\" border=\\"false\\" style=\\"width: 300px;\\"><!----><input class=\\"mb-select-view-input mb-select-view-input-hidden\\" readonly=\\"\\" placeholder=\\"Please select ...\\"><span class=\\"mb-select-view-value\\">Leaf</span><span class=\\"mb-select-view-suffix\\"><!----><span class=\\"mb-select-view-icon\\"><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"mb-icon mb-icon-down mb-select-view-arrow-icon\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\"><path d=\\"M39.6 17.443 24.043 33 8.487 17.443\\"></path></svg></span>
<!----></span></span>
<!---->"
`;

exports[`<tree-select> demo: render [load-more] correctly 1`] = `
"<span class=\\"mb-select-view-single mb-select-view mb-select-view-size-medium\\" border=\\"false\\" style=\\"width: 300px;\\"><!----><input class=\\"mb-select-view-input\\" readonly=\\"\\" placeholder=\\"Please select ...\\"><span class=\\"mb-select-view-value mb-select-view-value-hidden\\"><!----></span><span class=\\"mb-select-view-suffix\\"><!----><span class=\\"mb-select-view-icon\\"><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"mb-icon mb-icon-down mb-select-view-arrow-icon\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\"><path d=\\"M39.6 17.443 24.043 33 8.487 17.443\\"></path></svg></span>
<!----></span></span>
<!---->"
`;

exports[`<tree-select> demo: render [multiple] correctly 1`] = `
"<div class=\\"mb-space mb-space-horizontal mb-space-align-center\\">
  <div class=\\"mb-space-item\\" style=\\"margin-right: 8px;\\"><span class=\\"mb-select-view mb-select-view-size-medium mb-select-view-has-suffix mb-select-view-has-placeholder mb-select-view-multiple\\" border=\\"false\\" style=\\"width: 300px;\\"><span class=\\"mb-select-view-mirror\\">Please select ...</span>
    <!---->
    <transition-group-stub><input class=\\"mb-select-view-input\\" style=\\"width: 12px;\\" placeholder=\\"Please select ...\\"></transition-group-stub>
    <!----><span class=\\"mb-select-view-suffix\\"><!----><span class=\\"mb-select-view-icon\\"><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"mb-icon mb-icon-expand\\" style=\\"transform: rotate(-45deg);\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\"><path d=\\"M7 26v14c0 .552.444 1 .996 1H22m19-19V8c0-.552-.444-1-.996-1H26\\"></path></svg></span>
    <!---->
    <!----></span></span>
    <!---->
  </div>
  <div class=\\"mb-space-item\\"><span class=\\"mb-select-view mb-select-view-size-medium mb-select-view-has-suffix mb-select-view-has-placeholder mb-select-view-multiple\\" border=\\"false\\" style=\\"width: 300px;\\"><span class=\\"mb-select-view-mirror\\">Please select ...</span>
    <!---->
    <transition-group-stub><input class=\\"mb-select-view-input\\" style=\\"width: 12px;\\" placeholder=\\"Please select ...\\"></transition-group-stub>
    <!----><span class=\\"mb-select-view-suffix\\"><!----><span class=\\"mb-select-view-icon\\"><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"mb-icon mb-icon-expand\\" style=\\"transform: rotate(-45deg);\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\"><path d=\\"M7 26v14c0 .552.444 1 .996 1H22m19-19V8c0-.552-.444-1-.996-1H26\\"></path></svg></span>
    <!---->
    <!----></span></span>
    <!---->
  </div>
</div>"
`;

exports[`<tree-select> demo: render [popup-visible] correctly 1`] = `
"<div style=\\"margin-bottom: 24px;\\"><button class=\\"mb-btn mb-btn-primary mb-btn-shape-square mb-btn-size-medium mb-btn-status-normal\\" type=\\"button\\">
    <!--v-if-->toggle
  </button></div><span class=\\"mb-select-view-single mb-select-view mb-select-view-size-medium\\" border=\\"false\\" style=\\"width: 300px;\\"><!----><input class=\\"mb-select-view-input\\" readonly=\\"\\" placeholder=\\"Please select ...\\"><span class=\\"mb-select-view-value mb-select-view-value-hidden\\"><!----></span><span class=\\"mb-select-view-suffix\\"><!----><span class=\\"mb-select-view-icon\\"><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"mb-icon mb-icon-down mb-select-view-arrow-icon\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\"><path d=\\"M39.6 17.443 24.043 33 8.487 17.443\\"></path></svg></span>
<!----></span></span>
<!---->"
`;

exports[`<tree-select> demo: render [search] correctly 1`] = `
"<div class=\\"mb-space mb-space-horizontal mb-space-align-center\\">
  <div class=\\"mb-space-item\\" style=\\"margin-right: 8px;\\"><span class=\\"mb-select-view-single mb-select-view mb-select-view-size-medium mb-select-view-search\\" border=\\"false\\" style=\\"width: 300px;\\"><!----><input class=\\"mb-select-view-input\\" placeholder=\\"Please select ...\\"><span class=\\"mb-select-view-value mb-select-view-value-hidden\\"><!----></span><span class=\\"mb-select-view-suffix\\"><!----><span class=\\"mb-select-view-icon\\"><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"mb-icon mb-icon-expand\\" style=\\"transform: rotate(-45deg);\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\"><path d=\\"M7 26v14c0 .552.444 1 .996 1H22m19-19V8c0-.552-.444-1-.996-1H26\\"></path></svg></span>
    <!----></span></span>
    <!---->
  </div>
  <div class=\\"mb-space-item\\"><span class=\\"mb-select-view-single mb-select-view mb-select-view-size-medium mb-select-view-search\\" border=\\"false\\" style=\\"width: 300px;\\"><!----><input class=\\"mb-select-view-input\\" placeholder=\\"Please select ...\\"><span class=\\"mb-select-view-value mb-select-view-value-hidden\\"><!----></span><span class=\\"mb-select-view-suffix\\"><!----><span class=\\"mb-select-view-icon\\"><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"mb-icon mb-icon-expand\\" style=\\"transform: rotate(-45deg);\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\"><path d=\\"M7 26v14c0 .552.444 1 .996 1H22m19-19V8c0-.552-.444-1-.996-1H26\\"></path></svg></span>
    <!----></span></span>
    <!---->
  </div>
</div>"
`;

exports[`<tree-select> demo: render [search-remote] correctly 1`] = `
"<span class=\\"mb-select-view-single mb-select-view mb-select-view-size-medium mb-select-view-search\\" border=\\"false\\" style=\\"width: 300px;\\"><!----><input class=\\"mb-select-view-input\\" placeholder=\\"Please select ...\\"><span class=\\"mb-select-view-value mb-select-view-value-hidden\\"><!----></span><span class=\\"mb-select-view-suffix\\"><!----><span class=\\"mb-select-view-icon\\"><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"mb-icon mb-icon-expand\\" style=\\"transform: rotate(-45deg);\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\"><path d=\\"M7 26v14c0 .552.444 1 .996 1H22m19-19V8c0-.552-.444-1-.996-1H26\\"></path></svg></span>
<!----></span></span>
<!---->"
`;

exports[`<tree-select> demo: render [size] correctly 1`] = `
"<div style=\\"margin-bottom: 20px;\\"><span class=\\"mb-radio-group-button mb-radio-group-size-undefined mb-radio-group-direction-horizontal\\"><label class=\\"mb-radio-button\\"><input type=\\"radio\\" class=\\"mb-radio-target\\" value=\\"mini\\"><span class=\\"mb-radio-button-content\\">mini</span></label><label class=\\"mb-radio-button\\"><input type=\\"radio\\" class=\\"mb-radio-target\\" value=\\"small\\"><span class=\\"mb-radio-button-content\\">small</span></label><label class=\\"mb-radio-button mb-radio-checked\\"><input type=\\"radio\\" class=\\"mb-radio-target\\" value=\\"medium\\"><span class=\\"mb-radio-button-content\\">medium</span></label><label class=\\"mb-radio-button\\"><input type=\\"radio\\" class=\\"mb-radio-target\\" value=\\"large\\"><span class=\\"mb-radio-button-content\\">large</span></label></span></div><span class=\\"mb-select-view-single mb-select-view mb-select-view-size-medium\\" border=\\"false\\" style=\\"width: 300px;\\"><!----><input class=\\"mb-select-view-input mb-select-view-input-hidden\\" readonly=\\"\\" placeholder=\\"Please select ...\\"><span class=\\"mb-select-view-value\\">node1</span><span class=\\"mb-select-view-suffix\\"><!----><span class=\\"mb-select-view-icon\\"><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"mb-icon mb-icon-down mb-select-view-arrow-icon\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\"><path d=\\"M39.6 17.443 24.043 33 8.487 17.443\\"></path></svg></span>
<!----></span></span>
<!---->"
`;

exports[`<tree-select> demo: render [trigger-element] correctly 1`] = `
"<div class=\\"mb-typography\\" style=\\"width: 300px;\\"> You selected: <a href=\\"javascript: void(0)\\">node1</a>
  <!---->
  <!--v-if-->
  <!--v-if-->
  <!--v-if-->
</div>
<!---->"
`;

exports[`<tree-select> demo: render [virtual] correctly 1`] = `
"<span class=\\"mb-select-view-single mb-select-view mb-select-view-size-medium\\" border=\\"false\\"><!----><input class=\\"mb-select-view-input\\" readonly=\\"\\"><span class=\\"mb-select-view-value mb-select-view-value-hidden\\"><!----></span><span class=\\"mb-select-view-suffix\\"><!----><span class=\\"mb-select-view-icon\\"><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"mb-icon mb-icon-down mb-select-view-arrow-icon\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\"><path d=\\"M39.6 17.443 24.043 33 8.487 17.443\\"></path></svg></span>
<!----></span></span>
<!---->"
`;
